### DISCUSSION TITLE
defmulti confusion

### DISCUSSION THREAD

**Discussion by oskarkv** [hylang/hy#2023] [Q&A] [answered]
Hi!

I want a Clojure style `defmulti`. I read the docs at https://docs.hylang.org/en/test-branch-for-docs/contrib/multi.html but they seem outdated. There is no `hy.contrib.multi` anymore it seems. How can I make a Clojure style multimethod now?

**Comment by peaceamongworlds** [comment, marked as answer]
That link is for a very old and outdated version of the docs.

[This article](https://adambard.com/blog/implementing-multimethods-in-python/) seems to have be a good introduction to implementing multimethods in Python. The final constructs with functions and decorators can easily be wrapped in a macro if you want the same syntax as Clojure.

[This](https://github.com/hylang/hy/blob/0.17.0/hy/contrib/multi.hy) is Hy's implementation before it was removed.

**Reply by allison-casey** [reply]
There are also several pypi packages that implement multidispatch in a more pythonic way that you can use if you're looking for a drop in replacement if you don't want to roll your own.

**Reply by asemic-horizon** [reply]
I keep feeling there's a lack of clear communication on deprecated versions, which is the 0.x that's "final for now" until 1.0 drops on PyPI, and what are the differences between versions.

When you open the Python 2.7 docs by mistake, a big red banner shows up on the top saying "We don't support this anymore, please get with the program". Besides, the color scheme for old docs is completely different. This is not all too difficult with sphinx -- probably easy with whatever autodoc system Hylang is using.

**Reply by asemic-horizon** [reply]
Maybe easier still is if someone on the core team writes a *blog post* about what Hy has been in the past, why it's changing, what are the fixed points people can work with, etc. and post it somewhere like Hacker News. The risk in using Hy right now (I'm taking the risk, but I'm not particularly smart) is that this high-level vision isn't articulated anywhere. Here on the outside, no one really knows where this is going. Strong languages like Perl have waned under a much lower level of uncertainty.

**Reply by allison-casey** [reply]
It's a valid point and something I think i'd like to come back to once we get through the bikeshedding and implementation of the documentation overhaul i'll be doing soon.

**Reply by andrewgiessel** [reply]
Any brief update on this?  I'd really love to start using Hy for some projects, but am hesitant to do so until something like 1.0 comes along.

**Reply by allison-casey** [reply]
i have some time this weekend, i'll try and get a post up about the changes going into 1.0 and why for y'all

**Reply by allison-casey** [reply]
I posted and pinned an update for our drive to 1.0. check it out when you get the chance!
